diagrams/app_data_dependencies.rb

```diff
@@ -0,0 +1,100 @@
+require 'yaml'
+
+class AppDependencyGenerator
```

Comment:
I'm not sure how useful this is once we've done the initial update of `applications.yml`. Maybe it can output (not write) all the dependencies it thinks are missing?

Reply:
commented out code in this file